ui/src/pages/frontsite/PatronProfile/PatronCurrentLoans/components/ExtendButton/ExtendButton.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { DateTime } from 'luxon';
+import { Button, Icon, Popup } from 'semantic-ui-react';
+import { invenioConfig, ES_DELAY } from '@config';
+import _get from 'lodash/get';
+import _has from 'lodash/has';
+
+export const INFO_MESSAGES = {
+  hasMaxExtensions: 'You have reached the max number of extensions for a loan!',
+  hasExtendAction:
+    'Manually extending this loan, is not currently an available action!',
+  hasPendingLoans:
+    'Other users requested that book, therefore you cannot extend your loan.',
+  isOverdue: 'You loan is overdue, therefore you cannot extend the loan!',
+  isOwner: 'You are not the owner of this loan, so you cannot extend it!',
+};
+
+export default class ExtendButton extends Component {
+  state = {
+    hasExtendAction: false,
+    hasMaxExtensions: false,
+    hasPendingLoans: false,
+    infoMessage: '',
+    isOverdue: false,
+    isOwner: false,
+  };
+
+  componentDidMount = () => {
+    this.checkMaxExtensions();
```

Comment:
comment: I just want to stress out that these checks will run only when the component will be mounted. This means that if you depend on re running these after a prop change or a rerender they will not run :)